code/main.py

```diff
@@ -0,0 +1,53 @@
+'''
+Get data and models and train them
+'''
+
+import os
+import argparse
+from data import NlpData
+from models import NlpModel
+
+# Global Constants
+DIR = os.path.dirname(os.path.abspath(__file__))
+PARSER = argparse.ArgumentParser(description='LSTM Implementation in Tensorflow')
+
+def main(args):
+    data_train = NlpData(os.path.join(DIR, args.data_dir), args.file_name, args.vocab_size)
+    batch = data_train.get_next_batch(args.batch_size)
+    model = NlpModel(
+        args.batch_size, 
+        args.seq_length, 
+        args.embedding_size, 
+        args.hidden_state_size,
+        args.vocab_size,
+        args.summaries_dir)
+    # model.session.run() to access the tf.session variable
+
+if __name__ == '__main__':
+    PARSER.add_argument("--predef", help="Predefined mode for all arguments", type=int)
```

Comment:
Nitpick. http://stackoverflow.com/questions/5262702/argparse-module-how-to-add-option-without-any-argument